jina/serve/runtimes/worker/http_csp_app.py

```diff
@@ -154,50 +155,81 @@ async def post(request: Request):
                         detail='Invalid CSV input. Please check your input.',
                     )
 
-                def construct_model_from_line(
-                        model: Type[BaseModel], line: List[str]
-                ) -> BaseModel:
+
+                def construct_model_from_line(model: Type[BaseModel], line: List[str]) -> BaseModel:
+                    origin = get_origin(model)
+                    # If the model is of type Optional[X], unwrap it to get X
+                    if origin is Union:
+                        # If the model is of type Optional[X], unwrap it to get X
+                        args = get_args(model)
+                        if type(None) in args:
+                            model = args[0]
+
                     parsed_fields = {}
                     model_fields = model.__fields__
 
-                    for field_str, (field_name, field_info) in zip(
-                            line, model_fields.items()
-                    ):
-                        field_type = field_info.outer_type_
-
-                        # Handle Union types by attempting to arse each potential type
-                        if get_origin(field_type) is Union:
-                            for possible_type in get_args(field_type):
-                                if possible_type is str:
-                                    parsed_fields[field_name] = field_str
-                                    break
-                                else:
+                    for idx, (field_name, field_info) in enumerate(model_fields.items()):
+                        field_type = field_info.type_
+                        field_str = line[idx]  # Corresponding value from the row
+
+                        try:
+                            # Handle Literal types (e.g., Optional[Literal["value1", "value2"]])
+                            origin = get_origin(field_type)
+                            if origin is Literal:
+                                literal_values = get_args(field_type)
+                                if field_str not in literal_values:
+                                    raise HTTPException(
+                                        status_code=400,
```

Comment:
use thr fastapi status codes and not plain numbers

Reply:
done